xapian-core/tests/unittest.cc

```diff
@@ -881,6 +881,49 @@ static bool test_parseunsigned1()
     return true;
 }
 
+template<typename S>
+inline static void parsesigned_helper() {
+    S val;
+    const S max_val = numeric_limits<S>::max();
+    const S min_val = numeric_limits<S>::min();
+    tout << "Testing with parsesigned_helper" << endl;
+    TEST(parse_signed("0", val));
+    TEST_EQUAL(val, 0);
+    TEST(parse_signed("99", val));
+    TEST_EQUAL(val, 99);
+    TEST(parse_signed("-99", val));
+    TEST_EQUAL(val, -99);
+    TEST(parse_signed(str(max_val).c_str(), val));
+    TEST_EQUAL(val, max_val);
+    TEST(parse_signed(str(min_val).c_str(), val));
+    TEST_EQUAL(val, min_val);
+    TEST(!parse_signed("", val));
+    TEST(!parse_signed("abc", val));
+    TEST(!parse_signed("0a", val));
+    TEST(!parse_signed("-99a", val));
+    TEST(!parse_signed("-a99", val));
+    TEST(!parse_signed("--99", val));
+
+    unsigned long long temp = max_val + 1ull;
+    TEST(!parse_signed(str(temp).c_str(), val));
+
+    if (sizeof(S) < sizeof(long long)) {
+	long long temp1 = static_cast<long long>(min_val) - 1LL;
+	TEST(!parse_signed(str(temp1).c_str(), val));
+    }
+}
+
+static bool test_parsesigned1()
+{
+    parsesigned_helper<char>();
```

Comment:
Implementations are allow to make `char` signed or unsigned by default (it's signed on x86 Linux, but e.g. on arm Linux it is unsigned - https://wiki.debian.org/ArchitectureSpecificsMemo#Summary has a table for Debian architectures).  To get type which is always signed you need to use `signed char`.

Reply:
fixed in a fixup commit